hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -376,6 +380,180 @@ public static Schema createNewSchemaFromFieldsWithReference(Schema schema, List<
     return newSchema;
   }
 
+  /**
+   * If schemas are projection equivalent, then a record with schema1 does not need to be projected to schema2
+   * because the projection will be the identity.
+   *
+   *  Two schemas are considered projection equivalent if the field names and types are equivalent.
+   *  The names of records, namespaces, or docs do not need to match. Nullability is ignored.
+   */
+  public static boolean areSchemasProjectionEquivalent(Schema schema1, Schema schema2) {
```

Comment:
Does this method `areSchemasProjectionEquivalent` provide the same functionality as `AvroSchemaComparatorForSchemaEvolution#schemaEquals`?

Reply:
No. That one allows for type promotion.